sys/linux/sys.txt

```diff
@@ -89,12 +89,16 @@ read(fd fd, buf buffer[out], count len[buf])
 pread64(fd fd, buf buffer[out], count len[buf], pos fileoff)
 readv(fd fd, vec ptr[in, array[iovec_out]], vlen len[vec])
 preadv(fd fd, vec ptr[in, array[iovec_out]], vlen len[vec], off fileoff)
+preadv2(fd fd, vec ptr[in, array[iovec_out]], vlen len[vec], off fileoff, flags flags[rwf_flags])
```

Comment:
Humm.... looking at the kernel signature, it actually have `unsigned long, pos_l, unsigned long, pos_h`. and the same for preadv. I wonder if we pass args correctly for preadv/preadv2.
Please check with a debug print in the kernel code.

Reply:
That is right, and prevalent to `preadv`, `pwritev`, `preadv2`, `pwritev2`.

In raw syscalls, the offset is unpacked into two two args, which are `pos_l` and `pos_h`.

[Fixed](https://github.com/google/syzkaller/pull/1967/commits/31113f0382276057e2b2654fe8685ca5402cf86f).